app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+
+def create_person_list(people_list: list[dict[str, any]]) -> list[Person]:
+    new_person_list = [
+        Person(person.get("name"), person.get("age")) for person in people_list
+    ]
+
+    for person in people_list:
+        if person.get("wife") and person["wife"] in Person.people:
+            wife = Person.people[person.get("wife")]
+            Person.people[person.get("name")].wife = wife
+
+        if person.get("husband") and person["husband"] in Person.people:
+            husband = Person.people[person.get("husband")]
+            Person.people[person.get("name")].husband = husband
```

Comment:
The code assumes that the 'wife' and 'husband' keys in the dictionaries refer to existing `Person` objects. If the name does not exist in `Person.people`, this will raise a `KeyError`. Please add a check to ensure the name exists in `Person.people` before accessing it to prevent potential errors.